server/services/cms.js

```diff
@@ -65,64 +71,66 @@ class CmsService {
     }
   }
 
+  async getSuggestions(establishmentName, secondaryTags, categories) {
+    const limit = 4;
+    const secondaryTagSuggestions = await this.#cmsApi.get(
+      new SuggestionSecondaryTagQuery(
+        establishmentName,
+        secondaryTags.map(({ uuid }) => uuid),
+        limit,
+      ),
+    );
+    const categorySuggestions =
+      secondaryTagSuggestions.length < limit && categories
+        ? await this.#cmsApi.get(
+            new SuggestionCategoryQuery(
+              establishmentName,
+              categories.map(({ uuid }) => uuid),
+              limit,
+            ),
+          )
+        : [];
+    return [...secondaryTagSuggestions, ...categorySuggestions]
+      .slice(0, limit)
+      .filter(
+        ({ id }, i, self) =>
+          self.findIndex(({ id: findId }) => id === findId) === i,
+      );
```

Comment:
cannot read this, what's it doing?

Reply:
wow, ok... think its removing duplicates. Maybe could extract into a named function or something?